components/merkledb/src/proof_map_index/proof_builder.rs

```diff
@@ -158,22 +161,23 @@ impl ContourNode {
 /// implement `BuildProof` as well.
 ///
 /// [`MerklePatriciaTree`]: trait.MerklePatriciaTree.html
-pub trait BuildProof<K, V> {
+pub trait BuildProof<K, V, KeyMode> {
     /// Creates a proof of existence / absence for a single key.
-    fn create_proof(&self, key: K) -> MapProof<K, V>;
+    fn create_proof(&self, searched_path: ProofPath, key: K) -> MapProof<K, V, KeyMode>;
```

Comment:
Since `BuildProof` is parametric by `KeyMode`, converting keys to paths outside of it seems an overkill. This is especially true for the `create_multiproof` method, which may involve reallocation (or at least it seemingly does now, if the compiler doesn't eliminate it).